modules/drouting/drouting.c

```diff
@@ -2324,12 +2329,34 @@ static mi_response_t *mi_dr_get_partition(const mi_params_t *params,
 	return NULL;
 }
 
+static inline int get_inherit_state (const mi_params_t *params) {
+	str inherit_state;
+	int is_inherit_state = 1; // default is inhert old state
+
+	if (get_mi_string_param(params, "inherit_state", &inherit_state.s, &inherit_state.len) >= 0) {
+		LM_DBG("inherit_state is: %s \n", inherit_state.s);
+
+		if (inherit_state.s[0] == '0' || inherit_state.s[0] == 'n' || inherit_state.s[0] == 'N') {
+			is_inherit_state = 0;
+		}
+		else if (inherit_state.s[0] == '1' || inherit_state.s[0] == 'y' || inherit_state.s[0] == 'Y') {
+			is_inherit_state = 1;
+		} else {
+			LM_WARN("inherit_state values was not recognized, ignored, use default 1\n");
+		}
+	}
+
+	return is_inherit_state;
+}
+
```

Comment:
This is not necessarily related to this PR, but as this is 100% the same idea as in `dispatcher`, I would like to have some consistency, in terms of:

- using a function like `get_inherit_state` for evaluating the param value
- using enum or simply 0/1 values for the internal value of the inherit variable